ansible/roles/ocp4-workload-idm/tasks/workload.yml

```diff
@@ -0,0 +1,112 @@
+---
+# Implement your Workload deployment tasks here
+
+- name: Setting up workload for user
+  debug:
+    msg: "Setting up workload for user ocp_username = {{ ocp_username }}"
+
+# Find IP of bastion
+- name: Gather VPC facts
+  ec2_vpc_net_facts:
+    filters:
+      tag:Name: "{{ aws_vpc_name }}"
+    region: "{{ aws_region }}"
+  register: vpc
+
+- name: Get instance facts
+  ec2_instance_facts:
+    filters:
+# FIXME - find a better way to discover the bastion name
+      "tag:Name": "{{ instances[0].name }}"
+    region: "{{ aws_region }}"
+  register: instancesr
+  failed_when: instancesr|length == 0
+
+- name: Get non-terminated instance
+  set_fact:
+    ec2instance: "{{ item }}"
+  with_items:
+  - "{{ instancesr.instances }}"
+  when: not item.state.name == 'terminated'
+  failed_when: ec2instance is undefined
+
+- include_role:
+    name: config-idm-server
+  vars:
+    become_override: yes
+    private_ip: "{{ec2instance.private_ip_address}}"
```

Comment:
I'm not a big fan of roles including other roles. That could lead to dependencies hell.

Can we instead add `config-idm-server` to `infra_workloads` so it's loaded before ocp4-workload-idm ?

```yaml
infra_workloads:
  [...]
  - config-idm-server
  - ocp4-workload-idm
  [...]
```

Also i think that way it'd be self-documented that the bastion is going to serve as an idm-server.

You can probably guess the private_ip_address from the role itself.

Reply:
If I'm not mistaken infra-workloads runs after ocp4 workloads?  I thought the new paradigm was to use default_workloads for ocp4_workloads - see ocp4-workshop/env_vars.yml.  In general I thought we were moving away from infra_workloads with OCP 4?

Reply:
in order:

* default_workloads
* infra_workloads
* student_workloads

We can override `default_workloads` or use `infra_workloads`, it doesn't really matter.
We usually use `infra_workloads` for things more dynamic, for example to load workloads picked from a dialog in Cloudforms.

But my comment is still valid either it's `default_workloads` or `infra_workloads`

Reply:
ocp4-workloads-enable-lets-encrypt-certificates uses the technique of including a role so that the reusable code gets run at the precise time that it is needed in the workflow. I think this is a better solution than hoping that someone puts the workloads in the right order in their env_vars file.